addons/website_payment/static/tests/tours/donation.js

```diff
@@ -0,0 +1,122 @@
+            trigger: "p.alert-danger"
+        },
+        // -- End of testing the minimum amount --
+        {
+            content: "Enter a custom amount",
+            trigger: "#s_donation_amount_input",
+            run: "text 55"
+        },
+        {
+            content: "Donate with custom amount set",
+            trigger: ".s_donation_donate_btn",
+            run: "click"
+        },
+        {
+            content: "Check if custom amount radio input is selected and value is 55",
+            trigger: "input#other_amount",
+            run: function(action) {
+                const input = action.tip_widget.$anchor[0];
+                const inputValue = input.nextElementSibling.children[0].value;
+                if (!input.checked || inputValue !== "55.0") {
+                    console.error("The custom amount radio input is not selected or the value is not 55.");
+                }
+            }
+        },
+        {
+            content: "Select the amount of 25",
+            trigger: "input#amount_1",
+            run: "click"
+        },
+        {
+            content: "Verify that only one input is checked",
+            trigger: "input#amount_1",
+            run: function(action) {
+                const amount1 = action.tip_widget.$anchor[0];
+                const amountDiv = amount1.parentElement.parentElement;
+                const radiosInputs = amountDiv.querySelectorAll("input[type=radio]");
+                const selectedInputs = Array.from(radiosInputs).filter(radio => radio.checked);
+
+                if (selectedInputs.length !== 1) {
+                    console.error("More than one input is checked.");
+                }
+            }
+        },
+        {
+            content: "Change custom amount to 67",
+            trigger: "input#other_amount",
+            run: function(action) {
+                const input = action.tip_widget.$anchor[0];
+                const customInput = input.nextElementSibling.children[0];
+                customInput.value = "67";
+                customInput.dispatchEvent(new Event("input", { bubbles: true }));
+                customInput.dispatchEvent(new Event("change", { bubbles: true }));
+            }
+        },
+        {
+            content: "Select the custom amount radio button",
+            trigger: "input#other_amount",
+            run: "click"
+        },
+        {
+            content: "Submit the donation form",
+            trigger: "button[name='o_payment_submit_button'",
+            run: "click"
+        },
+        {
+            content: "Verify that the amount displayed is 67",
+            trigger: "span.oe_currency_value",
+            run: function(action) {
+                const amountElement = action.tip_widget.$anchor[0];
+                if (!amountElement || amountElement.textContent.trim() !== "67.00") {
+                    console.error("The amount displayed is not 67.00.");
+                }
+            },
```

Comment:
It should be possible to match with `[value="67.00"]`

Reply:
Unfortunately, this is a text inside a span, I don't think there is a CSS selector for that.
`<span class="oe_currency_value">67.00</span>
`

Reply:
Would `:contains(...)` work ?

Reply:
Oh didn't know about that, my bad.